sunpy/net/dataretriever/sources/stereo.py

```diff
@@ -0,0 +1,353 @@
+                chk_var += 1
+        if (chk_var == 2):
+            return True
+        return False
+
+
+class SWAVESClient(GenericClient):
+    """
+    Returns a list of URLS to SWAVES files corresponding to value of input timerange.
+    URL source: `http://stereo-ssc.nascom.nasa.gov/data/beacon/`.
+    Parameters
+    ----------
+    timerange: sunpy.time.TimeRange
+        time range for which data is to be downloaded.
+        Example value - TimeRange('2015-12-30 00:00:00','2015-12-31 00:01:00')
+    
+    Instrument: Fixed argument, case-insensitive = 'swaves'
+    
+    Source: Two arguments, case-insensitive - 'ahead' or 'behind'
+    
+    Examples
+    --------
+    >>> from sunpy.net import Fido
+    >>> from sunpy.net import attrs as a    
+    >>> results = Fido.search(a.Time('2008/3/20 17:00:00', '2008/3/25 17:15:00'), a.Instrument('swaves'), a.Source('ahead'))
+    >>> print(results)
+    [<Table length=5>
+     Start Time           End Time      Source Instrument
+       str19               str19         str5     str6   
+    ------------------- ------------------- ------ ----------
+    2008-03-20 00:00:00 2008-03-21 00:00:00  ahead     swaves
+    2008-03-21 00:00:00 2008-03-22 00:00:00  ahead     swaves
+    2008-03-22 00:00:00 2008-03-23 00:00:00  ahead     swaves
+    2008-03-23 00:00:00 2008-03-24 00:00:00  ahead     swaves
+    2008-03-24 00:00:00 2008-03-25 00:00:00  ahead     swaves]
+    """
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        returns list of urls corresponding to given TimeRange.
+        """
+        source = kwargs.get('source').lower()
+        START_DATE = datetime.datetime(2006, 10, 27)
+        if (timerange.start < START_DATE):
+            raise ValueError('Earliest date for which SWAVES data is available is {:%Y-%m-%d}'.format(START_DATE))        
+        prefix = 'http://stereo-ssc.nascom.nasa.gov/data/beacon/{source}/'
+        suffix = 'swaves/%Y/%m/ST{char}_LB_SWAVES_%Y%m%d.idlsave'
+        url_pattern = prefix + suffix
+        crawler = Scraper(url_pattern, source = source, char = source[0].upper())
+        if not timerange:
+            return []
+        result = crawler.filelist(timerange)
+        return result
+
+    def _makeimap(self):
+        """
+        Helper Function: used to hold information about source.
+        """
+        self.map_['instrument'] = 'swaves'
+
+    @classmethod
+    def _can_handle_query(cls, *query):
+        """
+        Answers whether client can service the query.
+        
+        Parameters
+        ----------
+        query : list of query objects
+        
+        Returns
+        -------
+        boolean: answer as to whether client can service the query
+        
+        """
+        chkattr = ['Time', 'Instrument', 'Source']
+        chklist = [x.__class__.__name__ in chkattr for x in query]
+        chk_var = 0
+        sources = ['ahead', 'behind']
```

Comment:
That's not the right syntax for [sources in VSO](http://sdac.virtualsolar.org/cgi/show_details?source=STEREO_A). Please, use `STEREO_A` or `STEREO_B`